capstone/src/main/java/com/google/sps/data/ReadData.java

```diff
@@ -85,75 +103,63 @@ public void readFile(FileReader file, String zipCode) {
+        entity.setProperty("date", date);
+        entity.setProperty("completionStatus", completionStatus);
+        entity.setProperty("gender", gender);
+        entity.setProperty("ageRange", ageRange);
+        entity.setProperty("directExperience", directExperience);
+        entity.setProperty("rating", rating);
+        entity.setProperty("text", text);
+        entity.setProperty("score", score);
+        entity.setProperty("responseTimeOne", responseTimeOne);
+        entity.setProperty("responseTimeTwo", responseTimeTwo);
+        entity.setProperty("responseTimeThree", responseTimeThree);
+
+        // If the survey maps to more than 1 precinct, average the precinct data
+        ArrayList<String> precinctNames = MapData.getPrecincts(zipCode);
+
+        int sumIncome = 0;
+        int sumCrimeRate = 0;
+        int sumStationRating = 0;
+        int total = 0;
+        for (String precinctName : precinctNames) {
+          Precinct precinct = FeatureData.getPrecinct(precinctName);
+
+          sumIncome += precinct.getAverageHouseholdIncome();
+          sumCrimeRate += precinct.getCrimeRate();
+          sumStationRating += precinct.getPoliceStationRating();
+
+          total++;
         }
 
-        if (inStore == null) {
-          entity.setProperty("zipCode", zipCode);
-          entity.setProperty("id", id);
-          entity.setProperty("date", date);
-          entity.setProperty("completionStatus", completionStatus);
-          entity.setProperty("gender", gender);
-          entity.setProperty("ageRange", ageRange);
-          entity.setProperty("directExperience", directExperience);
-          entity.setProperty("rating", rating);
-          entity.setProperty("text", text);
-          entity.setProperty("score", score);
-          entity.setProperty("responseTimeOne", responseTimeOne);
-          entity.setProperty("responseTimeTwo", responseTimeTwo);
-          entity.setProperty("responseTimeThree", responseTimeThree);
-
-          // If the survey maps to more than 1 precinct, average the precinct data
-          ArrayList<String> precinctNames = MapData.getPrecincts(zipCode);
-
-          int sumIncome = 0;
-          int sumCrimeRate = 0;
-          int sumStationRating = 0;
-          int total = 0;
-          for (String precinctName : precinctNames) {
-            Precinct precinct = FeatureData.getPrecinct(precinctName);
-
-            sumIncome += precinct.getAverageHouseholdIncome();
-            sumCrimeRate += precinct.getCrimeRate();
-            sumStationRating += precinct.getPoliceStationRating();
-
-            total++;
-          }
-
-          if (total != 0) {
-            entity.setProperty("averageHouseholdIncome", Math.round(sumIncome / total));
-            entity.setProperty("crimeRate", Math.round(sumCrimeRate / total));
-            entity.setProperty("policeStationRating", Math.round(sumStationRating / total));
-          }
-
-          newEntities.add(entity);
+        if (total != 0) {
```

Comment:
If total is 0 do we want to set the rating/crime rate/average income to something to indicate that there's not data?

Reply:
I believe it's set to null by default in data store

Reply:
Would we want to set those to 0 or null?

Reply:
Either works in this case since I check if the property is null